rules/RESPONSE-956-DATA-LEAKAGES-RUBY.conf

```diff
@@ -0,0 +1,108 @@
+# ------------------------------------------------------------------------
+# OWASP CRS ver.4.14.0-dev
+# Copyright (c) 2006-2020 Trustwave and contributors. All rights reserved.
+# Copyright (c) 2021-2025 CRS project. All rights reserved.
+#
+# The OWASP CRS is distributed under
+# Apache Software License (ASL) version 2
+# Please see the enclosed LICENSE file for full details.
+# ------------------------------------------------------------------------
+
+#
+# -= Paranoia Level 0 (empty) =- (apply unconditionally)
+#
+
+# Skip all rules if RESPONSE_BODY is compressed.
+SecRule RESPONSE_HEADERS:Content-Encoding "@pm gzip compress deflate br zstd" \
+    "id:956010,\
+    phase:4,\
+    pass,\
+    nolog,\
+    tag:'OWASP_CRS',\
+    tag:'OWASP_CRS/DATA-LEAKAGES-RUBY',\
+    ver:'OWASP_CRS/4.14.0-dev',\
+    skipAfter:END-RESPONSE-956-DATA-LEAKAGES-RUBY"
+
+SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 1" "id:956011,phase:3,pass,nolog,tag:'OWASP_CRS',ver:'OWASP_CRS/4.14.0-dev',skipAfter:END-RESPONSE-956-DATA-LEAKAGES-RUBY"
+SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 1" "id:956012,phase:4,pass,nolog,tag:'OWASP_CRS',ver:'OWASP_CRS/4.14.0-dev',skipAfter:END-RESPONSE-956-DATA-LEAKAGES-RUBY"
+#
+# -= Paranoia Level 1 (default) =- (apply only when tx.detection_paranoia_level is sufficiently high: 1 or higher)
+#
+
+#
+# -=[ RUBY Error Message Leakage ]=-
+#
+SecRule RESPONSE_BODY "@pmFromFile ruby-errors.data" \
+    "id:956100,\
+    phase:4,\
+    block,\
+    capture,\
+    t:none,\
+    msg:'RUBY Information Leakage',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}',\
+    tag:'application-multi',\
+    tag:'language-ruby',\
+    tag:'platform-multi',\
+    tag:'attack-disclosure',\
+    tag:'paranoia-level/1',\
+    tag:'OWASP_CRS',\
+    tag:'OWASP_CRS/DATA-LEAKAGES-RUBY',\
+    tag:'capec/1000/118/116',\
+    tag:'PCI/6.5.6',\
+    ver:'OWASP_CRS/4.14.0-dev',\
+    severity:'ERROR',\
+    setvar:'tx.outbound_anomaly_score_pl1=+%{tx.error_anomaly_score}'"
+
+
+# Detect the presence of the Ruby ERB templates "<%", "<%= " and slim interpolation "#{}" in output.
+#
+# To prevent false positives (due to the short "<%" sequences), we also include,
+# the space after it in an attempt to stop alerts in binary output.
+# And we make it case insensitive.
+#
+SecRule RESPONSE_BODY "@rx (?i)(?:<%=?\s+|#\{.*?\})" \
+    "id:956110,\
+    phase:4,\
+    block,\
+    capture,\
+    t:none,\
+    msg:'Ruby source code leakage',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}',\
+    tag:'application-multi',\
+    tag:'language-ruby',\
+    tag:'platform-multi',\
+    tag:'attack-disclosure',\
+    tag:'paranoia-level/1',\
+    tag:'OWASP_CRS',\
+    tag:'OWASP_CRS/DATA-LEAKAGES-RUBY',\
+    tag:'capec/1000/118/116',\
+    tag:'PCI/6.5.6',\
```

Comment:
This sounds wrong to me, according to https://pcidssguide.com/pci-dss-requirement-6/. I would expect `PCI/6.5`. @fzipi, @dune73?

Reply:
Could you open an issue for checking that? Because 6.5.6 is probably wrong everywhere. Then we don't need to address it in this PR.

Reply:
That is true. However, what is considered a high risk actually depends on the assessment of the people performing the risk review. They could also consider some forms of information leakage as low risk.